packages/gatsby-remark-autolink-headers/src/__tests__/index.js

```diff
@@ -182,4 +182,23 @@ describe(`gatsby-remark-autolink-headers`, () => {
       expect(node.data.id).toEqual(expect.stringMatching(/^heading/))
     })
   })
+
+  it(`extracts custom id`, () => {
+    const markdownAST = remark.parse(`
+# Heading One {#cusom_h1}
+
+## Heading Two {#custom-heading-two}
+
+# With *Bold* {#cusgom-withbold}
+    `)
+    const enableCustomId = true
+
+    const transformed = plugin({ markdownAST }, { enableCustomId })
+
+    visit(transformed, `heading`, node => {
+      expect(node.data.id).toBeDefined()
+
+      expect(node).toMatchSnapshot()
```

Comment:
Sort of a nit: but generally I'd argue we way overuse snapshots and they make future failures more confusing than they need to be.

Could you rather test for a specific id, e.g. something like

```
const customId = `custom-id`

const markdownAST = remark.parse(`
# Heading With a Custom Id {#${customId}}
`)

// later

visit(transformed, `heading`, node => {
  expect(node.data.id).toBe(customId)
})
```

Obviously you can scale this however you'd like if there's value in testing for multiple instances.

Reply:
I agree. I rewrote this test without using snapshots. Does it look better? If you like this, I will rewrite other tests with this style in another PR.